Kickstarter-iOS/Views/Transitions/SheetOverlayTransitionAnimator.swift

```diff
@@ -0,0 +1,99 @@
+    return 0.3
+  }
+
+  func animateTransition(using transitionContext: UIViewControllerContextTransitioning) {
+    guard let fromVC = transitionContext.viewController(forKey: .from),
+      let toVC = transitionContext.viewController(forKey: .to) else {
+        return
+    }
+
+    let containerView = transitionContext.containerView
+
+    if toVC.isBeingPresented {
+      self.animatePresentation(fromViewController: fromVC,
+                               toViewController: toVC,
+                               containerView: containerView,
+                               transitionContext: transitionContext)
+    } else {
+      self.animateDismissal(fromViewController: fromVC,
+                            toViewController: toVC,
+                            containerView: containerView,
+                            transitionContext: transitionContext)
+    }
+  }
+
+  private func animatePresentation(
+    fromViewController fromVC: UIViewController,
+    toViewController toVC: UIViewController,
+    containerView: UIView,
+    transitionContext: UIViewControllerContextTransitioning) {
+    _ = self.darkOverlayView
+      |> \.alpha .~ 0.0
+      |> \.frame .~ fromVC.view.frame
+
+    _ = toVC.view
+      |> \.backgroundColor .~ .clear
+      |> \.frame .~ fromVC.view.frame.offsetBy(dx: 0, dy: toVC.view.frame.height)
+
+    _ = (self.darkOverlayView, containerView)
+      |> ksr_addSubviewToParent()
+
+    _ = (toVC.view, containerView)
+      |> ksr_addSubviewToParent()
+
+    let toFrame = fromVC.view.frame
+
+    UIView.animate(
+      withDuration: self.transitionDuration(using: transitionContext),
+      delay: 0,
+      options: .curveEaseInOut,
+      animations: { [weak self] in
+        toVC.view.frame = toFrame
+        self?.darkOverlayView.alpha = 1.0
+    }, completion: { [weak self] _ in
+      toVC.view.backgroundColor = self?.darkOverlayView.backgroundColor
+      self?.darkOverlayView.removeFromSuperview()
+
+      transitionContext.completeTransition(!transitionContext.transitionWasCancelled)
+    })
+  }
+
+  private func animateDismissal(fromViewController fromVC: UIViewController,
+                                toViewController toVC: UIViewController,
+                                containerView: UIView,
+                                transitionContext: UIViewControllerContextTransitioning) {
+    _ = self.darkOverlayView
+      |> \.alpha .~ 1.0
+
+    _ = fromVC.view
+      |> \.backgroundColor .~ .clear
+
+    containerView.insertSubview(self.darkOverlayView, belowSubview: fromVC.view)
+
+    let toFrame = toVC.view.frame.offsetBy(dx: 0, dy: fromVC.view.frame.height)
+
+    UIView.animate(
+      withDuration: self.transitionDuration(using: transitionContext),
+      delay: 0,
+      options: .curveEaseInOut,
+      animations: { [weak self] in
+        fromVC.view.frame = toFrame
+        self?.darkOverlayView.alpha = 0.0
```

Comment:
I noticed that the `ProjectNavigatorTransitionAnimator` didn't use lenses to set animatable properties - not sure if this was intentional, it could be a performance thing?